### ISSUE TITLE
Client configuration should use OIDC auto-discovery

### ISSUE THREAD

**Issue by bbockelm** [ncsa/oa4mp#138]
Right now, a client config looks something like this:

```
    <client name="proxy-client">
        ... (some stuff) ....
        <serviceUri>https://cilogon.org/oauth2</serviceUri>
        <authorizeUri>https://cilogon.org/authorize</authorizeUri>
        <wellKnownUri>https://cilogon.org/oauth2/.well-known/openid-configuration</wellKnownUri>
        ... (other stuff) ....
    </client>
```

Where the `serviceUri` is then used to construct default URLs, such as the `accessTokenUri`, `authorizeUri`, `deviceAuthorizationUri`, and `userInfoUri` by assuming that the service for the client is following OA4MP's pattern of endpoint construction.

This could all be replaced with setting the issuer URL and using metadata discovery.

**Comment by jjg-123** [comment]
Agreed. This was put into place long before there was discovery. However, they still should be retained in case they need to be locally overridden.

**Comment by jjg-123** [comment]
Available in 5.5
